# Task-03

In [2]:
# Import necessary libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score, classification_report

# --- Data Loading with Error Handling ---
file_path = '/content/bank-full.csv'
try:
    bank_data = pd.read_csv(file_path, delimiter=';', error_bad_lines=False)
except Exception as e:
    print(f"An error occurred: {e}")

# Assuming the dataset is loaded successfully past the error handling above:
# --- Data Preprocessing ---
label_encoder = LabelEncoder()
categorical_columns = bank_data.select_dtypes(include=['object']).columns
for column in categorical_columns:
    bank_data[column] = label_encoder.fit_transform(bank_data[column])

# Split dataset into features (X) and target (y)
X = bank_data.drop('y', axis=1)
y = bank_data['y']

# Split data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# --- Model Building, Training, and Evaluation ---
dt_classifier = DecisionTreeClassifier(random_state=42)
dt_classifier.fit(X_train, y_train)
y_pred = dt_classifier.predict(X_test)

# Evaluation
accuracy = accuracy_score(y_test, y_pred)
print("Accuracy:", accuracy)
print(classification_report(y_test, y_pred))


<ipython-input-2-fa6b3b1c3faf>:11: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.


  bank_data = pd.read_csv(file_path, delimiter=';', error_bad_lines=False)


Accuracy: 0.8740462235983634
              precision    recall  f1-score   support

           0       0.93      0.93      0.93      7952
           1       0.48      0.48      0.48      1091

    accuracy                           0.87      9043
   macro avg       0.70      0.70      0.70      9043
weighted avg       0.87      0.87      0.87      9043



**Interpretation and Insights:**

**Data Preprocessing:** We encoded categorical variables to numeric, making the dataset suitable for machine learning algorithms. The dataset was then split into features and target variables, followed by splitting into training and testing sets.

**Model Building and Training:** We built a Decision Tree classifier, a straightforward yet powerful model for classification tasks. The model was trained on the training data without any hyperparameter tuning for simplicity.

**Model Evaluation: **The Decision Tree achieved an accuracy of approximately 87.4% on the test data. It performed well in predicting non-subscribers (class 0) with high precision, recall, and F1-score. However, its performance in predicting subscribers (class 1) was less accurate, indicating potential areas for improvement.